.github/workflows/ubuntu-rubygems.yml

```diff
@@ -44,6 +44,9 @@ jobs:
       - name: Run Test
         run: rake test
         if: "!startsWith(matrix.ruby.name, 'truffleruby') && !startsWith(matrix.ruby.name, 'jruby')"
+      - name: Run Test isolatedly
+        run: rake test:isolated
+        if: "!startsWith(matrix.ruby.name, 'truffleruby') && !startsWith(matrix.ruby.name, 'jruby')"
```

Comment:
We may not need to run this test in all the CRubies. Running on the latest stable CRuby might be enough good like this.

```
diff --git a/.github/workflows/ubuntu-rubygems.yml b/.github/workflows/ubuntu-rubygems.yml
index 5bc7ff8365..29cf6440f7 100644
--- a/.github/workflows/ubuntu-rubygems.yml
+++ b/.github/workflows/ubuntu-rubygems.yml
@@ -46,7 +46,7 @@ jobs:
         if: "!startsWith(matrix.ruby.name, 'truffleruby') && !startsWith(matrix.ruby.name, 'jruby')"
       - name: Run Test isolatedly
         run: rake test:isolated
-        if: "!startsWith(matrix.ruby.name, 'truffleruby') && !startsWith(matrix.ruby.name, 'jruby')"
+        if: "!startsWith(matrix.ruby.name, '3.1')"
       - name: Run Test (JRuby)
         run: JRUBY_OPTS=--debug rake test
         if: startsWith(matrix.ruby.name, 'jruby')
```